scripts/dappnode_install_pre.sh

```diff
@@ -20,25 +19,33 @@ detect_installation_type() {
     fi
 }
 
-
 add_docker_repo() {
     apt-get update -y
     apt-get remove -y docker docker-engine docker.io containerd runc | tee -a $LOG_FILE
-    apt-get install -y ca-certificates curl gnupg lsb-release | tee -a $LOG_FILE
-    mkdir -p /etc/apt/keyrings && chmod -R 0755 /etc/apt/keyrings
-    curl -fsSL "https://download.docker.com/linux/${lsb_dist}/gpg" | gpg --dearmor --yes -o /etc/apt/keyrings/docker.gpg
+
+    # Add Docker GPG key
+    apt-get install -y ca-certificates curl lsb-release | tee -a $LOG_FILE
+    install -m 0755 -d /etc/apt/keyrings
+    curl -fsSL "https://download.docker.com/linux/${lsb_dist}/gpg" -o /etc/apt/keyrings/docker.asc
     chmod a+r /etc/apt/keyrings/docker.gpg
-    echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/$lsb_dist $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+
+    # Add the repository to APT sources
+    echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/$lsb_dist $(lsb_release -cs) stable" |
+        tee /etc/apt/sources.list.d/docker.list >/dev/null
+
+    apt-get update -y
 }
 
 # DOCKER INSTALLATION
 install_docker() {
     apt-get update -y
-    apt-get install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin | tee -a $LOG_FILE
+    apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin | tee -a $LOG_FILE
 
     # Ensure xz is installed
-    [ -f "/usr/bin/xz" ] || ( apt-get install -y xz-utils)
+    [ -f "/usr/bin/xz" ] || (apt-get install -y xz-utils)
 
+    # Not working in Ubuntu ISO because the user is not created before executing late-commands
```

Comment:
if it is not working in ubuntu and has to be done in `dappnode_install.sh`, shouldnt we only add user in docker group only if NOT in ubuntu? to avoid unnecessary warning or errors

Reply:
In Ubuntu it will just not be executed because the user with ID `1000` does not exist yet.